linux/guest_customization/check_network_config.yml

```diff
@@ -108,28 +108,40 @@
   debug: var=guest_gateway_success
 
 # Collect network config files for non-Photon Linux
+# Ubuntu Desktop 19.10 and later, Ubuntu Server 17.10 and later use netplan to manage network
 - block:
-    - name: "Set fact of the network configure file for Ubuntu server"
+    - name: "Set traditional GOSC network configuration file on Ubuntu {{ guest_os_ansible_distribution_ver }}"
       set_fact:
-        src_network_file: "{{ netplan_config_file }}"
-      when:
-        - guest_os_ansible_distribution == "Ubuntu"
-        - guest_os_with_gui is defined and not guest_os_with_gui
-        - netplan_config_file is defined
-    - block:
-        - name: "Set traditional GOSC network configuration files on Ubuntu Desktop/Debian OS"
-          set_fact:
-            src_network_file: "/etc/network/interfaces"
-          when: "gosc_workflow == 'perl'"
-
-        - name: "Set cloud-init GOSC network configuration files on Ubuntu/Debian OS"
-          set_fact:
-            src_network_file: "/etc/network/interfaces.d/50-cloud-init.cfg"
-          when: "gosc_workflow == 'cloud-init'"
-      when: >
-        (guest_os_ansible_distribution == "Ubuntu" and guest_os_with_gui is defined and guest_os_with_gui) or
-        (guest_os_ansible_distribution != "Ubuntu")
-  when: guest_os_family == "Debian"
+        src_network_file: "/etc/netplan/99-netcfg-vmware.yaml"
```

Comment:
What about the netplan config file got from this task, is it not applicable here? And the file recorded here can cover all the versions?
```
# Get netplan config file for Ubuntu server    
- include_tasks: ../utils/get_netplan_config_file.yml
```

Reply:
Take Ubuntu Server as an example,  get_netplan_config_file.yml returns 00-installer-config.yaml, which is the original network config file. But GOSC will add a new network config file: 99-netcfg-vmware.yaml for perl GOSC and 50-cloud-init.yaml for cloud-init GOSC. So, here we don't use get_netplan_config_file.yml

Reply:
Ok, if it's the same on all the Ubuntu versions.